### ISSUE TITLE
Non exaustive pattern match error assumes a case expression

### ISSUE THREAD

**Issue by brodo** [elm-lang/error-message-catalog#58]
Hi.

```elm
myHead (head::tail) = head
myHead []
```

yields:
> Error: Runtime error in module Repl (on line 4, column 23 to 27)
> Non-exhaustive pattern match in case-expression.
> Make sure your patterns cover every case!

**Comment by mgold** [comment]
In the 0.16 night-before-prerelease, I see,

```
> myHead (head::tail) = head
==================================== ERRORS ====================================

-- PARTIAL PATTERN ------------------------------------------- repl-temp-000.elm

This pattern does not cover all possible inputs.

4│ myHead (head::tail) = head
           ^^^^^^^^^^
You need to account for the following values:

    []

Switch to a `case` expression to handle all possible patterns.

If you are seeing this error for the first time, check out these hints:
<https://github.com/elm-lang/elm-compiler/blob/0.16.0/hints/missing-patterns.md>
The recommendations about wildcard patterns and `Debug.crash` are important!
```

The link is wrong, I emailed Evan about it so he can fix it before the release, but other than that the error message seems to be much improved.

**Comment by rtfeldman** [comment]
Yep, this has been fixed (including the link) in the 0.16 release!
